registry/api/errcode/errors.go

```diff
@@ -97,9 +104,21 @@ func (e Error) Error() string {
 		e.Code.Message())
 }
 
-// Message returned the human-readable error message for this Error
-func (e Error) Message() string {
-	return e.Code.Message()
+// AddDetail will add some Detail info to the Error
+func (e Error) AddDetail(detail interface{}) Error {
```

Comment:
I think we may need a way to distinguish between adding more into to an existing Error and asking for a dup of the existing Errpr plus the new info. The pattern I was going with was.... WithXXX==Dup,   AddXXX==add to current.
This will avoid creating a set of temporary Errors that serve no purpose except as temporary ones.